### PR TITLE
Replace webpack with snowpack

### PR DESCRIPTION
**Pull request by manzt**
Just a bit of weekend fun. I came across the source code for [observable/plot](https://github.com/observablehq/plot), and thought it would be fun to experiment with a similar dev configuration. I ended up with something that can replace the current webpack config entirely, so I figured I'd open a PR if it is of any interest at all.

The main thing I was frustrated with is that (to my knowledge) you can't tell snowpack to treat `.fs` or `.vs` as JS modules (without writing a custom plugin), so I had to add `.js` extensions... 

Snowpack expects a 1:1 mapping of files, so I wrote a tiny watcher to generate each HTML endpoint when you edit `example/template.html` (previously `index.html`). This will run during production and dev environments. 

Again, this is all for fun so no worries if it's not of any interest! Regardless, I'd recommend checking out the branch and playing around with it :)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -124,7 +124,7 @@ import {
   rgbBrightness,
 } from './utils';
 
-import { version } from '../package.json';
+const version = import.meta.env.version;
```
Comment (manzt) [inline src/index.js:127]: https://github.com/snowpackjs/snowpack/discussions/1262

File: snowpack.config.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    fs.mkdirSync(outDir);
+  }
+  fs.readdirSync('example').forEach((file) => {
+    if (exclude.has(file)) return;
+    const outfile = path.join(outDir, file.replace('.js', '.html'));
+    fs.writeFileSync(
+      outfile,
+      html.replace(
+        '<!-- INSERT_ENTRYPOINT -->',
+        `<script type="module" src="%PUBLIC_URL%${file}"></script>`
```
Comment (manzt) [inline snowpack.config.js:25]: Inserts a script tag with the corresponding js entrypoint. "%PUBLIC_URL%" is resolved via snowpack config below: https://www.snowpack.dev/reference/configuration#buildoptionsbaseurl

File: src/index.js (hunk 1 above)
Comment (flekschas) [reply src/index.js:127]: Is this `import.meta.env` from snowpack? This looks a bit mystic to me and I can't say I like having build-tools specific imports in the main source code.

**Review by flekschas — COMMENTED** [review]
Wow thanks a lot! I was thinking to swap out webpack as well but never got around to it.

Two questions:
- Is there really no way Snowpack can treat other file extensions as JS imports? I named the shader files `.vs` and `.fs` to get syntax highlighting. Loosing syntax highlighting because of the example build process would be a show stopper I think.
- What are your thoughts on Snowpack vs Vite? Svelte moved away from Snowpack and chose Vite. I also know Vite is using esbuild under the hood, which is outrageously fast.

File: src/index.js (hunk 1 above)
Comment (manzt) [reply src/index.js:127]: I'm not at computer now but just had an idea of how to avoid this and keep imports the same!

**Comment by manzt** [comment]
> Is there really no way Snowpack can treat other file extensions as JS imports? I named the shader files .vs and .fs to get syntax highlighting. Loosing syntax highlighting because of the example build process would be a show stopper I think.

There is probably a way to do this with a snowpack [plugin](https://www.snowpack.dev/reference/plugins), I just wanted to get something working. I can explore this. 

> What are your thoughts on Snowpack vs Vite? Svelte moved away from Snowpack and chose Vite. I also know Vite is using esbuild under the hood, which is outrageously fast.

Vite and snowpack are very similar and both use `esbuild` under the hood. There is a good list of comparisons here: https://vitejs.dev/guide/comparisons.html. However that is a bit dated, and snowpack didn't have a builtin production story when it was written. 

TBH I tried configuring both for this project and just was able to get snowpack working first. Each have features/configurations that the others do not, but snowpack's ability to "mount" multiple directories at the website "root" works better for the current project structure.

**Pull request closed without merging**